recursion/core/src/multi/mod.rs

```diff
@@ -189,6 +223,8 @@ where
         let mut sub_builder = MultiBuilder::new(
             builder,
             local_multi_cols.is_fri_fold.into(),
+            builder.is_first_row(),
```

Comment:
Should we also add the condition local_multi_cols.is_fri (to handle the case when there are no fri rows).